modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormAdminActionDropdownItemsProviderTest.java

```diff
@@ -0,0 +1,567 @@
+
+		_assertActionDropdownItemEdit(dropdownGroupItems.get(0));
+		_assertActionDropdownItemShare(dropdownGroupItems.get(1));
+	}
+
+	private static void _setUpJSONFactoryUtil() {
+		JSONFactoryUtil jsonFactoryUtil = new JSONFactoryUtil();
+
+		jsonFactoryUtil.setJSONFactory(new JSONFactoryImpl());
+	}
+
+	private static void _setUpLanguageUtil() {
+		LanguageUtil languageUtil = new LanguageUtil();
+
+		Language language = mock(Language.class);
+
+		when(
+			language.get(
+				Mockito.any(HttpServletRequest.class), Mockito.anyString())
+		).thenAnswer(
+			invocation -> invocation.getArguments()[1]
+		);
+
+		languageUtil.setLanguage(language);
+	}
+
+	private void _assertActionDropdownItemDelete(DropdownItem dropdownItem) {
+		Assert.assertEquals(
+			HashMapBuilder.<String, Object>put(
+				"action", "delete"
+			).put(
+				"deleteFormInstanceURL",
+				StringBundler.concat(
+					"http//localhost/test?param_redirect=", _CURRENT_URL,
+					";param_javax.portlet.action=/dynamic_data_mapping_form",
+					"/delete_form_instance;param_formInstanceId=",
+					_FORM_INSTANCE_ID)
+			).build(),
+			dropdownItem.get("data"));
+		Assert.assertEquals("delete", dropdownItem.get("label"));
+		Assert.assertEquals("times-circle", dropdownItem.get("symbolLeft"));
+	}
+
+	private void _assertActionDropdownItemDuplicate(DropdownItem dropdownItem) {
+		Assert.assertEquals(
+			_INVALID_DDM_FORM_INSTANCE, dropdownItem.get("disabled"));
+		Assert.assertEquals(
+			StringBundler.concat(
+				"http//localhost/test?param_redirect=", _CURRENT_URL,
+				";param_javax.portlet.action=/dynamic_data_mapping_form",
+				"/copy_form_instance;param_groupId=", _SCOPE_GROUP_ID,
+				";param_formInstanceId=", _FORM_INSTANCE_ID),
+			dropdownItem.get("href"));
+		Assert.assertEquals("duplicate", dropdownItem.get("label"));
+		Assert.assertEquals("copy", dropdownItem.get("symbolLeft"));
+	}
+
+	private void _assertActionDropdownItemEdit(DropdownItem dropdownItem) {
+		Assert.assertEquals(
+			_INVALID_DDM_FORM_INSTANCE, dropdownItem.get("disabled"));
+		Assert.assertEquals(
+			StringBundler.concat(
+				"http//localhost/test?param_mvcRenderCommandName=/admin",
+				"/edit_form_instance;param_redirect=", _CURRENT_URL,
+				";param_formInstanceId=", _FORM_INSTANCE_ID),
+			dropdownItem.get("href"));
+		Assert.assertEquals("edit", dropdownItem.get("label"));
+		Assert.assertEquals("pencil", dropdownItem.get("symbolLeft"));
+	}
+
+	private void _assertActionDropdownItemExport(DropdownItem dropdownItem) {
+		Assert.assertEquals(
+			_INVALID_DDM_FORM_INSTANCE, dropdownItem.get("disabled"));
+		Assert.assertEquals(
+			StringBundler.concat(
+				"javascript:", _PORTLET_NAMESPACE,
+				"exportFormInstance('http//localhost/test?",
+				"param_formInstanceId=", _FORM_INSTANCE_ID, "');"),
+			dropdownItem.get("href"));
+		Assert.assertEquals("export", dropdownItem.get("label"));
+	}
+
+	private void _assertActionDropdownItemPermissions(
```

Comment:
As offline discussed, can we make sure that the items that don't verify if the form is invalid will always be enabled?

Reply:
Done!